provisioning/device/devdoc/dps_error_requirements.md

```diff
@@ -0,0 +1,31 @@
+# azure-iot-provisioning-device.translateError Requirements
+
+## Overview
+`translateError` is a method that translates DPS errors into Azure IoT Hub errors, effectively abstracting the error that is returned to the SDK user of from the transport layer.
+
+## Requirements
+
+** SRS_NODE_DPS_ERRORS_18_001: [** `translateError` shall accept 4 arguments:
+ - A custom error message to give context to the user.
+ - the status code that initiated the error
+ - the response body
+ - the response object itself **]**
+
+
+** SRS_NODE_DPS_ERRORS_18_008: [** Any error object returned by `translateError` shall inherit from the generic `Error` Javascript object and have 3 properties:
+ - `response` shall contain the `IncomingMessage` object returned by the HTTP layer.
+ - `reponseBody` shall contain the content of the HTTP response.
+ - `message` shall contain a human-readable error message **]**
+
+** SRS_NODE_DPS_ERRORS_18_002: [** `translateError` shall return an `ArgumentError` if the status code is `400`. **]**
+
+** SRS_NODE_DPS_ERRORS_18_003: [** `translateError` shall return an `UnauthorizedError` if the status code is `401`. **]**
+
+** SRS_NODE_DPS_ERRORS_18_004: [** `translateError` shall return an `DeviceNotFoundError` if the status code is `404`. **]**
+
+** SRS_NODE_DPS_ERRORS_18_005: [** `translateError` shall return an `IotHubQuotaExceededError` if the status code is `429`. **]**
```

Comment:
>IotHubQuotaExceededError [](start = 69, length = 24)

mhhh is that the same quota as the IoT Hub quota? might be misleading? do we need a ProvisioningQuotaExceededError or RegistrationQuotaExceededError?

Reply:
maybe, but I think we're splitting hairs here.

---
In reply to: [154874443](https://github.com/Azure/azure-iot-sdk-node/pull/169#discussion_r154874443) [](ancestors = 154874443)